src/services/db/GitFileSystemService.ts

```diff
@@ -453,12 +487,17 @@ export default class GitFileSystemService {
 
           return this.getGitBlobHash(repoName, path)
             .orElse(() => okAsync(""))
-            .andThen((sha) => {
+            .andThen((sha) =>
+              combine([okAsync(sha), this.getFilePathStats(repoName, path)])
+            )
+            .andThen((shaAndStats) => {
+              const [sha, stats] = shaAndStats as [string, fs.Stats]
               const result: GitDirectoryItem = {
                 name,
                 type,
                 sha,
                 path,
+                size: type === "dir" ? 0 : stats.size,
```

Comment:
Quick qn: How do I reason about the `size` key and why is this significant to the functionality of GGs?

Reply:
The size is the file size (or 0 for directories). It is not particularly for GGs but rather for compatibility with the return type of `GitHubService.readDirectory()`.